services/motion/builtin/builtin_test.go

```diff
@@ -1731,25 +1277,906 @@ func TestMoveOnGlobe(t *testing.T) {
-	test.That(t, err.Error(), test.ShouldEqual, "unimplemented")
-	test.That(t, executionID, test.ShouldResemble, uuid.Nil)
+		timeoutCtx, timeoutFn = context.WithTimeout(ctx, time.Second*5)
+		defer timeoutFn()
+		err = motion.PollHistoryUntilSuccessOrError(timeoutCtx, ms, time.Millisecond*5, motion.PlanHistoryReq{
+			ComponentName: req.ComponentName,
+			ExecutionID:   executionID,
+			LastPlanOnly:  true,
+		})
+		test.That(t, err, test.ShouldBeNil)
+
+		endPos, err := kb.CurrentPosition(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		logger.Debug(spatialmath.PoseToProtobuf(endPos.Pose()))
+		test.That(t, spatialmath.PoseAlmostEqualEps(endPos.Pose(), goal1BaseFrame, 10), test.ShouldBeTrue)
+
+		// Now, we try to go to the second goal. Since the `CurrentPosition` of our base is at `goal1`, the pose that motion solves for and
+		// logs should be {x:-1043  y:593}
+		req = motion.MoveOnMapReq{
+			ComponentName: base.Named("test-base"),
+			Destination:   goal2SLAMFrame,
+			SlamName:      slam.Named("test_slam"),
+			Extra:         map[string]interface{}{"smooth_iter": 5},
+		}
+		timeoutCtx, timeoutFn = context.WithTimeout(ctx, time.Second*5)
+		defer timeoutFn()
+		executionID, err = ms.(*builtIn).MoveOnMapNew(timeoutCtx, req)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, executionID, test.ShouldNotResemble, uuid.Nil)
+
+		timeoutCtx, timeoutFn = context.WithTimeout(ctx, time.Second*5)
+		defer timeoutFn()
+		err = motion.PollHistoryUntilSuccessOrError(timeoutCtx, ms, time.Millisecond*5, motion.PlanHistoryReq{
+			ComponentName: req.ComponentName,
+			ExecutionID:   executionID,
+			LastPlanOnly:  true,
+		})
+		test.That(t, err, test.ShouldBeNil)
+
+		endPos, err = kb.CurrentPosition(ctx)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, spatialmath.PoseAlmostEqualEps(endPos.Pose(), goal2BaseFrame, 5), test.ShouldBeTrue)
+
+		plans, err := ms.PlanHistory(ctx, motion.PlanHistoryReq{
+			ComponentName: base.Named("test-base"),
+			LastPlanOnly:  false,
+			ExecutionID:   executionID,
+		})
+		test.That(t, err, test.ShouldBeNil)
+
+		goalPose1 := plans[0].Plan.Steps[0][base.Named("test-base")]
+		goalPose2 := spatialmath.PoseBetween(
+			plans[0].Plan.Steps[0][base.Named("test-base")],
+			plans[0].Plan.Steps[len(plans[0].Plan.Steps)-1][base.Named("test-base")],
+		)
+
+		// We don't actually surface the internal motion planning goal; we report to the user in terms of what the user provided us.
+		// Thus, we use PlanHistory to get the plan steps of the latest plan.
+		// The zeroth index of the plan steps is the relative position of goal1 and the pose inverse between the first and last value of
+		// plan steps gives us the relative pose we solved for goal2.
+		test.That(t, spatialmath.PoseAlmostEqualEps(goalPose1, goal1BaseFrame, 10), test.ShouldBeTrue)
+
+		// This is the important test.
+		test.That(t, spatialmath.PoseAlmostEqualEps(goalPose2, spatialmath.PoseBetween(goal1BaseFrame, goal2BaseFrame), 10), test.ShouldBeTrue)
```

Comment:
@biotinker I changed to test to not search through the log files anymore. What do you think of this method of testing?

Reply:
This should work, yes. Nice work on this.

This test does require that the plan produced actually match the goal requested, but if that assumption does not hold then other tests should catch that, so IMO this is fine.